libpod/container_internal.go

```diff
@@ -976,6 +976,19 @@ func (c *Container) completeNetworkSetup() error {
 			}
 		}
 	}
+	// check if we have a bindmount for /etc/hosts
+	hostsBindMount := state.BindMounts["/etc/hosts"]
+	if hostsBindMount != "" && len(c.cniHosts()) > 0 {
```

Comment:
This should also only be done if `c.config.PostConfigureNetNS` is set.

Reply:
This check happened above
	if !c.config.PostConfigureNetNS || netDisabled {
		return nil
	}
So this will not happen if c.config.PostConfigureNetNS is not set or net was disabled.